client/src/components/GraphViewerComponent/GraphViewerCytoscapeComponent/GraphViewerCytoscapeComponent.js

```diff
@@ -565,10 +566,34 @@ export class GraphViewerCytoscapeComponent extends React.Component {
     });
   }
 
+  saveSessionLayout() {
+    const cy = this.graphControl;
+    const el = cy.nodes("*");
+    const currentLayoutPositions = sessionService.getCurrentGraphLayoutPositions(this.layout, this.query);
+    if (Object.keys(currentLayoutPositions).length === 0) {
+      el.forEach(node => {
+        sessionService.saveGraphLayoutNodesPosition(this.layout, this.query, node.data("id"), node.position("x"), node.position("y"));
+      });
+    }
+  }
+
+  loadSessionLayout() {
+    const cy = this.graphControl;
+    const el = cy.nodes("*");
+    el.forEach(node => {
+      const position = sessionService.getGraphLayoutNodesPosition(this.layout, this.query, node.data("id"));
+      node.position(position);
+    });
+  }
+
   setLayout(layout) {
     this.layout = layout;
   }
 
+  setQuery(query) {
+    this.query = query;
+  }
```

Comment:
This seems a bit weird. I.e. for the cytoscape component to know what the query is. It doesn't actually care about the query, it's just a cache key... plus it's also another method that's easy to forget about.

Could we update the loadSessionLayout and saveSessionLayout methods to take a new parameter `query`? Or `cacheKey`? And pass it then? Rather than storing a copy in this class?